build.gradle

```diff
@@ -34,6 +33,9 @@ subprojects {
         apply plugin: 'maven-publish'
         apply from: "$rootDir/gradle/publishing.gradle"
     } else {
+        java {
+            toolchain.languageVersion = JavaLanguageVersion.of(17)
```

Comment:
I take that needing JDK17 comes from the use of jfxtras-17? I'm okay having this condition here.

In any case, I see that even if I set my `JAVA_HOME` to 11, when running the samples, gradle manages to find a 17 SDK from my system and uses it, overriding my settings, which I'm not sure I like...

Also, if I set the toolchain with an SDK I don't have, for instance 18, I get `No locally installed toolchains match and toolchain download repositories have not been configured`. I see there is a plugin that can be added that will download "one", and install it under ~/.gradle/jdks, but again, I don't like this being done without user's control.

Reply:
> I take that needing JDK17 comes from the use of jfxtras-17? I'm okay having this condition here.

Yes

Reply:
> Also, if I set the toolchain with an SDK I don't have, for instance 18, I get No locally installed toolchains match and toolchain download repositories have not been configured. 

I believe toolchains will only download LTS/GA releases it can find a download for. My guess is that for 18 it can't find where to download it base on your system that satisfies those constraints.